### PR TITLE
Fix constant handling

### PR DESCRIPTION
**Pull request by VeaaC**
Adds support for:
- signed values
- normalized generator output (decimal always)
- width checking (enough bits for value?)

### CODE REVIEW

File: tests/tree/test_node_tree_builder.py [hunk 1]
```diff
@@ -94,13 +94,13 @@ def test_constants_are_placed_on_final_namespace():
 
 def test_constants_are_parsed():
     tree = _build_node_tree("""namespace foo{
-        const i8 foo = 17;
+        const i8 foo = -17;
```
Comment (boxdot) [inline tests/tree/test_node_tree_builder.py]: Could you please also add a test for negative hex value?

Comment (VeaaC) [reply tests/tree/test_node_tree_builder.py]: Done
